### DISCUSSION TITLE
How to setup in Docker on Synology with qBT?

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [cross-seed/cross-seed#349] [General]
I've been trying to set this up on my Synology using Docker for a really long time. I'm able to set it up on my Mac and run it remotely no problem, but I just can't figure out how to make it work on the Synology itself. I want to be able to use qBT's "Run external program on torrent completion" option, but after following the Wiki, seeing the container in the Synology Docker GUI, setting up the config, including port, every time I try to run it, the container just immediately shuts itself down for no discernible reason.

Also, I just learned that it over-writes my config file when I try to run it, and I have to redo it. I run several Docker containers, but I've never used Docker Compose, so maybe that's the key, but there are no instructions in the Wiki on how/where that goes, etc (in the config folder? outside of it? Help? haha).

Just to test, I tried to run the "curl" command that qBT is supposed to execute after connecting to my Syno via SSH, but it's telling me the port is blocked... but it's not, internally anyways, that I can tell. Anyone set this up on Synology in Docker and know how to help me?  Would love to not have to rely on another computer to be on and auto-running a Folder Action when a download finishes, and I can't seem to find a tutorial anywhere.

Thanks!

[UPDATE: I've added TZ, PGID, and PUID to my container environmental variables, but it still just automatically shuts the container down with no info. Tried opening all ports in the Synology firewall as well, still says it's blocked. No clue.]

**Comment by mmgoodnow** [comment]
Without providing any information about what you're doing that's not working, it'll be difficult for anybody to be able to help you. I highly recommend joining the Discord and asking about it there, we have a few users on there running cross-seed on Synology and the synchronous nature will help you get your answers faster. https://discord.com/invite/jpbUFzS5Wb

**Comment by SawkeeReemo** [comment]
I actually forgot that I made this post 8 months ago. 
But I ended up answering this same question for myself on another post here: https://github.com/cross-seed/cross-seed/discussions/496
